### ISSUE TITLE
quick start commands error when returning schedule

### ISSUE THREAD

**Issue by borfaxer** [rsheftel/pandas_market_calendars#287]
The commands in the Quick Start documentation produce an error. This was working a few weeks ago. Did something get changed about the underlying data type in Pandas that needs to be adapted to?

>>> import pandas_market_calendars as mcal
>>> nyse = mcal.get_calendar('NYSE')
>>> early = nyse.schedule(start_date='2012-09-05', end_date='2012-09-12')

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.10/dist-packages/pandas_market_calendars/market_calendar.py", line 616, in schedule
    special = self.special_dates(market_time, start_date, end_date, filter_holidays= False)
  File "/usr/local/lib/python3.10/dist-packages/pandas_market_calendars/market_calendar.py", line 557, in special_dates
    special = self._special_dates(calendars, ad_hoc, start_date, end_date)
  File "/usr/local/lib/python3.10/dist-packages/pandas_market_calendars/market_calendar.py", line 531, in _special_dates
    indexes = [
  File "/usr/local/lib/python3.10/dist-packages/pandas_market_calendars/market_calendar.py", line 532, in <listcomp>
    self.days_at_time(self._tryholidays(calendar, start, end), time_)
  File "/usr/local/lib/python3.10/dist-packages/pandas_market_calendars/market_calendar.py", line 521, in _tryholidays
    try: return cal.holidays(s, e)
  File "/usr/local/lib/python3.10/dist-packages/pandas/tseries/holiday.py", line 476, in holidays
    pre_holidays = [
  File "/usr/local/lib/python3.10/dist-packages/pandas/tseries/holiday.py", line 477, in <listcomp>
    rule.dates(start, end, return_name=True) for rule in self.rules
  File "/usr/local/lib/python3.10/dist-packages/pandas/tseries/holiday.py", line 283, in dates
    holiday_dates = self._apply_rule(dates)
  File "/usr/local/lib/python3.10/dist-packages/pandas/tseries/holiday.py", line 360, in _apply_rule
    return dates.map(lambda d: self.observance(d))
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py", line 6418, in map
    new_values = self._map_values(mapper, na_action=na_action)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/base.py", line 919, in _map_values
    return arr.map(mapper, na_action=na_action)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/arrays/_mixins.py", line 80, in method
    return meth(self, *args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/arrays/datetimelike.py", line 722, in map
    result = map_array(self, mapper, na_action=na_action)
  File "/usr/local/lib/python3.10/dist-packages/pandas/core/algorithms.py", line 1812, in map_array
    return lib.map_infer(values, mapper, convert=convert)
  File "lib.pyx", line 2917, in pandas._libs.lib.map_infer
  File "/usr/local/lib/python3.10/dist-packages/pandas/tseries/holiday.py", line 360, in <lambda>
    return dates.map(lambda d: self.observance(d))
  File "/usr/local/lib/python3.10/dist-packages/pandas_market_calendars/holidays_nyse.py", line 306, in july_5th_holiday_observance
    return datetime_index[datetime_index.year < 2013]
TypeError: 'Timestamp' object is not subscriptable

**Comment by rsheftel** [comment]
This works fine for me and all the tests pass. (1) try updating to the latest 4.3.0 release. (2) can you provide what package versions you are running with "pip list"

**Comment by borfaxer** [comment]
This was in a container I built the day I filed the issue, so I assumed it would be the latest release, but apparently not:
pandas-market-calendars 4.2.1
The container base is ubuntu 22.04, the most recent Long-Term Stable one. When I ran
pip3 install pandas_market_calendars --upgrade
it updated to 4.3.0 and is now working. Thank you! Strange that my initial build of the container 3 weeks earlier worked fine but the rebuild this week didn't.
